compiler/src/openxla/compiler/nvgpu/Dialect/CUDNN/Conversion/ConvertCUDNNToRuntime.cpp

```diff
@@ -663,6 +685,43 @@ struct ConvertCudnnBiasOp : public CudnnOpConversionPattern<BiasOp> {
   }
 };
 
+//===----------------------------------------------------------------------===//
+// cudnn.relu
+//===----------------------------------------------------------------------===//
+
+struct ConvertCudnnReluOp : public CudnnOpConversionPattern<ReluOp> {
+  using CudnnOpConversionPattern::CudnnOpConversionPattern;
+
+  LogicalResult matchAndRewrite(
+      ReluOp op, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+    MLIRContext *ctx = rewriter.getContext();
+    ImplicitLocOpBuilder b(op->getLoc(), rewriter);
+    SmallVector<Value> args(adaptor.getOperands());
+    auto f32 = rewriter.getF32Type();
+    int64_t alignment = op.getComputeType().getIntOrFloatBitWidth();
+    // It is unable to convert the lowerClip directly to float, thus the static
+    // cast.
+    float lower_clip =
+        static_cast<float>(adaptor.getLowerClip().convertToDouble());
+    args.push_back(b.create<arith::ConstantFloatOp>(APFloat(lower_clip), f32));
```

Comment:
You don't need to materialize it as float, you can construct args directly: https://github.com/openxla/openxla-nvgpu/blob/a243da3d8337c3f8cbe2f93f2165dd1abbdd011c/compiler/src/openxla/compiler/nvgpu/Dialect/CUDNN/Conversion/ConvertCUDNNToRuntime.cpp#L583

Reply:
I did the following below and I got the following crash:
```
args.push_back(b.create<arith::ConstantFloatOp>(op.getLowerClip(), f32));
```

```
$ ./iree_core/tools/iree-opt --iree-plugin=openxla-cudnn --split-input-file  --openxla-nvgpu-convert-cudnn-to-runtime test.mlir
<unknown>:0: error: FloatAttr type doesn't match the type implied by its value
iree-opt: <snip>/iree/third_party/llvm-project/mlir/include/mlir/IR/StorageUniquerSupport.h:174: static ConcreteT mlir::detail::StorageUserBase<mlir::FloatAttr, mlir::Attribute, mlir::detail::FloatAttrStorage, mlir::detail::AttributeUniquer, mlir::TypedAttr::Trait>::get(mlir::MLIRContext *, Args...) [ConcreteT = mlir::FloatAttr, BaseT = mlir::Attribute, StorageT = mlir::detail::FloatAttrStorage, UniquerT = mlir::detail::AttributeUniquer, Traits = <mlir::TypedAttr::Trait>, Args = <mlir::Type, llvm::APFloat>]: Assertion `succeeded(ConcreteT::verify(getDefaultDiagnosticEmitFn(ctx), args...))' failed.
Please report issues to https://github.com/openxla/iree/issues and include the crash backtrace.
Stack dump:
...
```
It doesn't like it to be `f32`. If I make it a `f64` it works fine.

It fails the same way if I replace the above line with this:

```
b.create<arith::ConstantFloatOp>(adaptor.getLowerClip(), f32));
```

Reply:
When I worked on this actually running anything through IREE failed with f64 and I could only make it work with f32.  As for the `FloatAttr` error, I think you just have to use `APFloat::IEEEsingle`.

Reply:
`F64Attr` in `Ops.td` should all be changed to `F32` until demote-f64-to-f32 is fixed

Reply:
I kept this as-is due to the issue Fredrik mentioned. When I fix that issue, I will go back and do this fix as Eugene mentioned.